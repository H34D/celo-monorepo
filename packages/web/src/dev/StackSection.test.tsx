import * as React from 'react'
import { I18nextProvider } from 'react-i18next'
import { Text } from 'react-native'
import * as renderer from 'react-test-renderer'
import StackSection from 'src/dev/StackSection'
import NextI18NextInstance from 'src/i18n'
import { ScreenSizeProvider } from 'src/layout/ScreenSize'

describe('StackSection', () => {
  it('renders with typical props', () => {
    const tree = renderer.create(
      <ScreenSizeProvider>
        <I18nextProvider i18n={NextI18NextInstance.i18n} initialLanguage="en">
          <StackSection
            title="Test"
            text="Stack Section Test"
            label="Driven Test Development"
            id="1"
            children={<Text>I am a Child</Text>}
            buttonOne={{ title: 'test', href: '/text' }}
            buttonTwo={{ title: 'test', href: '/text' }}
          />
        </I18nextProvider>
      </ScreenSizeProvider>
    )
    expect(tree).toMatchSnapshot()
  })
})
