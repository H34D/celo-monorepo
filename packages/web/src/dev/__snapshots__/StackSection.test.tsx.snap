// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackSection renders with typical props 1`] = `
<div
  className="react-reveal"
  style={
    Object {
      "opacity": undefined,
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    id="1"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignSelf": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignSelf": "center",
          "flexDirection": "row",
          "marginBottom": "60px",
          "maxWidth": "1100px",
          "msFlexDirection": "row",
          "msFlexItemAlign": "center",
          "width": "100%",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxFlex": 0,
            "WebkitFlexGrow": 0,
            "WebkitFlexShrink": 0,
            "flexGrow": 0,
            "flexShrink": 0,
            "msFlexNegative": 0,
            "msFlexPositive": 0,
            "paddingBottom": "10px",
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "paddingTop": "10px",
            "width": "25%",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "color": "rgba(255,255,255,1.00)",
              "fontFamily": "futura-pt, sans-serif",
              "fontSize": "72px",
              "lineHeight": "72px",
              "textRendering": "geometricPrecision",
            }
          }
        >
          Driven Test Development
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxFlex": 0,
            "WebkitBoxPack": "justify",
            "WebkitFlexGrow": 0,
            "WebkitFlexShrink": 0,
            "WebkitJustifyContent": "space-between",
            "flexGrow": 0,
            "flexShrink": 0,
            "justifyContent": "space-between",
            "msFlexNegative": 0,
            "msFlexPack": "justify",
            "msFlexPositive": 0,
            "paddingBottom": "10px",
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "paddingTop": "10px",
            "width": "50%",
          }
        }
      >
        <h3
          aria-level="3"
          className="css-reset-4rbku5 css-text-901oao"
          dir="auto"
          role="heading"
          style={
            Object {
              "color": "rgba(255,255,255,1.00)",
              "fontFamily": "futura-pt, sans-serif",
              "fontSize": "28px",
              "lineHeight": "32px",
              "marginBottom": "20px",
              "textRendering": "geometricPrecision",
              "textTransform": "none",
            }
          }
        >
          Test
        </h3>
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "color": "rgba(255,255,255,1.00)",
              "fontFamily": "eb-garamond, serif",
              "fontSize": "20px",
              "lineHeight": "28px",
              "textRendering": "geometricPrecision",
            }
          }
        >
          Stack Section Test
        </div>
        <ul
          className="css-reset-4rbku5 css-view-1dbjc4n"
          role="list"
          style={
            Object {
              "marginLeft": "10px",
              "marginTop": "20px",
              "paddingLeft": "20px",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
          >
            I am a Child
          </div>
        </ul>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxFlex": 0,
            "WebkitFlexGrow": 0,
            "WebkitFlexShrink": 0,
            "flexGrow": 0,
            "flexShrink": 0,
            "msFlexNegative": 0,
            "msFlexPositive": 0,
            "paddingBottom": "10px",
            "paddingLeft": "10px",
            "paddingRight": "10px",
            "paddingTop": "10px",
            "width": "25%",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onResponderGrant={[Function]}
          onResponderRelease={[Function]}
        >
          <a
            className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao"
            data-focusable={true}
            dir="auto"
            href="/text"
            role="link"
            style={
              Object {
                "WebkitAlignItems": "center",
                "WebkitBoxAlign": "center",
                "WebkitBoxFlex": 0,
                "WebkitBoxPack": "center",
                "WebkitFlexGrow": 0,
                "WebkitJustifyContent": "center",
                "alignItems": "center",
                "backgroundColor": "rgba(53,208,127,1.00)",
                "borderBottomColor": "rgba(53,208,127,1.00)",
                "borderBottomLeftRadius": "3px",
                "borderBottomRightRadius": "3px",
                "borderBottomWidth": "1px",
                "borderLeftColor": "rgba(53,208,127,1.00)",
                "borderLeftWidth": "1px",
                "borderRightColor": "rgba(53,208,127,1.00)",
                "borderRightWidth": "1px",
                "borderTopColor": "rgba(53,208,127,1.00)",
                "borderTopLeftRadius": "3px",
                "borderTopRightRadius": "3px",
                "borderTopWidth": "1px",
                "color": "rgba(255,255,255,1.00)",
                "cursor": "pointer",
                "display": "inline-flex",
                "flexGrow": 0,
                "fontFamily": "futura-pt, sans-serif",
                "fontSize": "16px",
                "fontWeight": "500",
                "justifyContent": "center",
                "lineHeight": "16px",
                "minWidth": "100px",
                "msFlexAlign": "center",
                "msFlexPack": "center",
                "msFlexPositive": 0,
                "opacity": 1,
                "paddingBottom": "7px",
                "paddingLeft": "10px",
                "paddingRight": "10px",
                "paddingTop": "7px",
                "textAlign": "center",
                "textRendering": "geometricPrecision",
                "width": "max-content",
              }
            }
            target="blank"
          >
            test
          </a>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "marginBottom": "10px",
              "marginTop": "10px",
            }
          }
        />
        <div
          className="css-view-1dbjc4n"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onResponderGrant={[Function]}
          onResponderRelease={[Function]}
          style={
            Object {
              "WebkitAlignItems": "flex-start",
              "WebkitBoxAlign": "start",
              "alignItems": "flex-start",
              "msFlexAlign": "start",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "borderBottomColor": "rgba(221,221,221,1.00)",
                "borderBottomLeftRadius": "3px",
                "borderBottomRightRadius": "3px",
                "borderBottomWidth": "1px",
                "borderLeftColor": "rgba(221,221,221,1.00)",
                "borderLeftWidth": "1px",
                "borderRightColor": "rgba(221,221,221,1.00)",
                "borderRightWidth": "1px",
                "borderTopColor": "rgba(221,221,221,1.00)",
                "borderTopLeftRadius": "3px",
                "borderTopRightRadius": "3px",
                "borderTopWidth": "1px",
              }
            }
          >
            <a
              className="css-reset-4rbku5 css-cursor-18t94o4 css-text-901oao"
              data-focusable={true}
              dir="auto"
              href="/text"
              role="link"
              style={
                Object {
                  "WebkitAlignItems": "center",
                  "WebkitBoxAlign": "center",
                  "WebkitBoxFlex": 0,
                  "WebkitBoxPack": "center",
                  "WebkitFlexGrow": 0,
                  "WebkitJustifyContent": "center",
                  "alignItems": "center",
                  "color": "rgba(53,208,127,1.00)",
                  "cursor": "pointer",
                  "display": "inline-flex",
                  "flexGrow": 0,
                  "fontFamily": "futura-pt, sans-serif",
                  "fontSize": "16px",
                  "fontWeight": "500",
                  "justifyContent": "center",
                  "lineHeight": "16px",
                  "minWidth": "100px",
                  "msFlexAlign": "center",
                  "msFlexPack": "center",
                  "msFlexPositive": 0,
                  "opacity": 1,
                  "paddingBottom": "6px",
                  "paddingLeft": "10px",
                  "paddingRight": "10px",
                  "paddingTop": "6px",
                  "textAlign": "center",
                  "textRendering": "geometricPrecision",
                  "width": "max-content",
                }
              }
              target="blank"
            >
              test
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
